File: src/main/java/nz/co/searchwellington/queues/RabbitConnectionFactory.java
package nz.co.searchwellington.queues;

import java.io.IOException;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import com.rabbitmq.client.Connection;
import com.rabbitmq.client.ConnectionFactory;

@Component
public class RabbitConnectionFactory {

	private final ConnectionFactory factory;
	
	@Autowired
	public RabbitConnectionFactory(
			@Value("${rabbit.hostname}") String rabbitHost,
			@Value("${rabbit.port}") Integer rabbitPort
	) {
		this.factory = new ConnectionFactory();
		factory.setHost(rabbitHost);
		factory.setPort(rabbitPort);
	}
	
	public Connection connect() throws IOException {
		return factory.newConnection();
	}
	
}

File: src/main/scala/nz/co/searchwellington/queues/RabbitConnectionFactory.scala
package nz.co.searchwellington.queues

import com.rabbitmq.client.{Connection, ConnectionFactory}
import org.springframework.beans.factory.annotation.{Autowired, Value}
import org.springframework.stereotype.Component

@Component
class RabbitConnectionFactory @Autowired()(@Value("${rabbit.hostname}") val rabbitHost: String,
                                           @Value("${rabbit.port}") val rabbitPort: Integer) {

  private val factory = {
    val factory = new ConnectionFactory
    factory.setHost(rabbitHost)
    factory.setPort(rabbitPort)
    factory
  }

  def connect: Connection = factory.newConnection

}